src/engine/SCons/CacheDir.py

```diff
@@ -139,16 +139,78 @@ def CachePushFunc(target, source, env):
 class CacheDir(object):
 
     def __init__(self, path):
+        """
+        Initialize a CacheDir object.
+
+        The cache configuration is stored in the object. It
+        is read from the config file in the supplied path if
+        one exists,  if not the config file is created and
+        the default config is written, as well as saved in the object.
+        """
         self.path = path
         self.current_cache_debug = None
         self.debugFP = None
         self.config = dict()
         if path is None:
             return
-        # See if there's a config file in the cache directory. If there is,
-        # use it. If there isn't, and the directory exists and isn't empty,
-        # produce a warning. If the directory doesn't exist or is empty,
-        # write a config file.
+
+        from SCons.Util import PY3
```

Comment:
Promote to top of file?

Reply:
Can do. Or could turn into a slightly wonky feature test instead, based on:

```
Python 2.7.15 (default, Oct 15 2018, 15:26:09) 
[GCC 8.2.1 20180801 (Red Hat 8.2.1-2)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.makedirs("foo", exist_ok=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: makedirs() got an unexpected keyword argument 'exist_ok'
>>> 
```